app/controllers/application_controller.rb

```diff
@@ -14,9 +14,9 @@ class ApplicationController < ActionController::Base
 
   protect_from_forgery with: :reset_session # See ActionController::RequestForgeryProtection for details
 
-  # Disable CSRF protection for non xml requests.
+  # Disable CSRF protection for requests to REST API.
   skip_before_action :verify_authenticity_token, if: -> do
-    (params.key?(:provider_key) || params.key?(:access_token)) && request.format.xml?
+    (params.key?(:provider_key) || params.key?(:access_token)) && api_controller?
```

Comment:
probably worthless micro-optimization, but shouldn't we first check `api_controller?` and then the other checks?